### PR TITLE
Display more detailed error to admin and lectors

### PR DESCRIPTION
**Pull request by MTsatsarov**
Closes [https://github.com/SoftUni-Internal/exam-systems-issues/issues/719](url)

New Exception types were added to help with debugging eventual exceptions.
Should catch every error and display detailed information on the UI if you are admin or lector

NOTE: There is another PR inside the workers related to this issue.

### CODE REVIEW

File: Open Judge System/Data/OJS.Data/Migrations/202306071340433_ExectuionCommentAndExceptionType.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+﻿namespace OJS.Data.Migrations
+{
+    using System;
+    using System.Data.Entity.Migrations;
+    
+    public partial class ExectuionCommentAndExceptionType : DbMigration
+    {
+        public override void Up()
+        {
+            AddColumn("dbo.Submissions", "ExecutionComment", c => c.String());
```
Comment (gogo4ds) [inline Open Judge System/Data/OJS.Data/Migrations/202306071340433_ExectuionCommentAndExceptionType.cs:10]: We are using automatic migrations in the old Judge, so no need to add manual migration. Do test it first though, as I haven't made automatic migrations recently..

File: Open Judge System/OJS.Common/Models/ExceptionType.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿namespace OJS.Common.Models
+{
+    public enum ExceptionType
+    {
+        None = 0,
+        Compile = 1,
+        Strategy = 2,
+        Runtime = 3,
+        Remote = 4,
+    }
```
Comment (gogo4ds) [inline Open Judge System/OJS.Common/Models/ExceptionType.cs:10]: Better use the enums from the workers repo, instead of duplicating them.

**Review by gogo4ds — CHANGES_REQUESTED** [review]

File: Open Judge System/Web/OJS.Web/Areas/Contests/Views/Submissions/View.cshtml [hunk 3]
```diff
@@ -79,6 +79,13 @@
     {
         <h2>@Resource.Execution_result:</h2>
         <pre>@Model.ProcessingComment</pre>
+        if (Model.UserHasAdminPermission)
+        {
+            <pre>@Resource.Exception_type: @Model.ExceptionType  
+            @Model.ExecutionComment
+        </pre>
```
Comment (mishogv) [inline Open Judge System/Web/OJS.Web/Areas/Contests/Views/Submissions/View.cshtml:86]: Format correctly

**Review by mishogv — CHANGES_REQUESTED** [review]

**Review by gogo4ds — DISMISSED** [review]

**Comment by gogo4ds** [comment]
Can you link the related PR in this one?

**Review by gogo4ds — CHANGES_REQUESTED** [review]

**Review by mishogv — APPROVED** [review]

**Review by gogo4ds — DISMISSED** [review]

**Review by gogo4ds — APPROVED** [review]

**Pull request merged**